lib/AST/PlatformKind.cpp

```diff
@@ -263,15 +263,43 @@ bool swift::inheritsAvailabilityFromPlatform(PlatformKind Child,
   return false;
 }
 
-llvm::VersionTuple swift::canonicalizePlatformVersion(
-    PlatformKind platform, const llvm::VersionTuple &version) {
-
-  // Canonicalize macOS version for macOS Big Sur to treat
-  // 10.16 as 11.0.
-  if (platform == PlatformKind::macOS ||
-      platform == PlatformKind::macOSApplicationExtension) {
-    return llvm::Triple::getCanonicalVersionForOS(llvm::Triple::MacOSX,
-                                                  version);
+std::optional<llvm::Triple::OSType>
+swift::tripleOSTypeForPlatform(PlatformKind platform) {
+  switch (platform) {
+  case PlatformKind::macOS:
+  case PlatformKind::macOSApplicationExtension:
+    return llvm::Triple::MacOSX;
+  case PlatformKind::iOS:
+  case PlatformKind::iOSApplicationExtension:
+  case PlatformKind::macCatalyst:
+  case PlatformKind::macCatalystApplicationExtension:
+    return llvm::Triple::IOS;
+  case PlatformKind::tvOS:
+  case PlatformKind::tvOSApplicationExtension:
+    return llvm::Triple::TvOS;
+  case PlatformKind::watchOS:
+  case PlatformKind::watchOSApplicationExtension:
+    return llvm::Triple::WatchOS;
+  case PlatformKind::visionOS:
+  case PlatformKind::visionOSApplicationExtension:
+    return llvm::Triple::XROS;
+  case PlatformKind::OpenBSD:
+    return llvm::Triple::OpenBSD;
```

Comment:
nit: what about FreeBSD? We've been adding support recently.

CC @michael-yuji

Reply:
This is a covered switch; there isn't a FreeBSD `PlatformKind` yet. I don't know whether or not there should be one but it's pretty orthogonal to this PR.